src/core/ActionRegister.h

```diff
@@ -100,14 +109,53 @@ std::ostream & operator<<(std::ostream &log,const ActionRegister&ar);
 /// \param classname the name of the class to be registered
 /// \param directive a string containing the corresponding directive
 /// This macro should be used in the .cpp file of the corresponding class
+/**
 #define PLUMED_REGISTER_ACTION(classname,directive) \
   namespace { class PLUMED_UNIQUENAME(classname##RegisterMe){ \
     static std::unique_ptr<PLMD::Action> create(const PLMD::ActionOptions&ao){return PLMD::Tools::make_unique<classname>(ao);} \
   public: \
     PLUMED_UNIQUENAME(classname##RegisterMe)(){PLMD::actionRegister().add(directive,create,classname::registerKeywords);} \
     ~PLUMED_UNIQUENAME(classname##RegisterMe)(){PLMD::actionRegister().remove(create);} \
   } PLUMED_UNIQUENAME(classname##RegisterMe); }
+* /
+namespace PLMD{
+//namespace actionSetup{
+template<typename ActionType>
+void myregisterAction(const std::string& key){
+  PLMD::actionRegister().add(key,
+  [](const PLMD::ActionOptions&ao)->std::unique_ptr<PLMD::Action> {
+    return PLMD::Tools::make_unique<ActionType>(ao);
+    },
+    ActionType::registerKeywords);
+}
+//}
+}*/
+/*
+#define PLUMED_REGISTER_ACTION(classname,directive) \
+  template<>\
+  ::PLMD::PLMD::actionRegister()<classname>(directive);
+*/
+//in C++20 you we'll make this a concept a change typename in plumedRegisterer
+template<typename T>
+inline constexpr bool isActionType = std::is_base_of<::PLMD::Action, T>::value;
+//concept ActionType = std::is_base_of<::PLMD::Action, T>::value;
+//and the next template will be template<ActionType ActionType>
+
+template<typename ActionClass>
+class plumedRegisterer{ 
```

Comment:
We usually use capital letters for classes (PlumedRegisterer)

Reply:
i corrected that in the fourth commit, is that not up?

Reply:
Maybe I was reviewing an older commit sorry for this